subsys/bluetooth/host/iso.c

```diff
@@ -705,29 +705,45 @@ static int bt_iso_setup_data_path(struct bt_conn *conn)
 		.dir = BT_HCI_DATAPATH_DIR_CTLR_TO_HOST };
 	struct bt_iso_data_path in_path = {
 		.dir = BT_HCI_DATAPATH_DIR_HOST_TO_CTLR };
+	struct bt_iso_chan_io_qos *tx_qos;
+	struct bt_iso_chan_io_qos *rx_qos;
 
 	chan = SYS_SLIST_PEEK_HEAD_CONTAINER(&conn->channels, chan, node);
 	if (!chan) {
 		return -EINVAL;
 	}
 
-	in_path.path = chan->qos->tx->path ? chan->qos->tx->path : &path;
-	out_path.path = chan->qos->rx->path ? chan->qos->rx->path : &path;
+	tx_qos = chan->qos->tx;
+	rx_qos = chan->qos->rx;
 
-	if (!chan->qos->tx) {
+	in_path.path = tx_qos && tx_qos->path ? tx_qos->path : &path;
+	out_path.path = rx_qos && rx_qos->path ? rx_qos->path : &path;
+
+	if (!tx_qos) {
 		in_path.pid = BT_ISO_DATA_PATH_DISABLED;
 	}
 
-	if (!chan->qos->rx) {
+	if (!rx_qos) {
 		out_path.pid = BT_ISO_DATA_PATH_DISABLED;
 	}
 
-	err = hci_le_setup_iso_data_path(conn, &in_path);
-	if (err) {
-		return err;
-	}
+	if (conn->iso.is_bis) {
+		if (tx_qos) {
```

Comment:
For the unitiated the direction of the path and it's relation to in/out path looks strange, is this behavior properly documented in the API?

Reply:
That's a very good question. I think it's fairly well documented that if `tx` is NULL then you cannot transmit, and if `rx` is NULL you cannot receive, but if you think there is something missing there, please let us know.

Reply:
I don't understand the behavior when the application does not initialize the path in QoS.

```
	/** Channel data path reference.
	 *  Setting to NULL default to HCI data path.
	 */
	struct bt_iso_chan_path		*path;
```

What does "default to HCI data path" mean?

The other thing is the how the NULL of tx_qos/rx_qos behaves for BIS.
The code sets up only one direction based off TX_QOS is NULL or not.
So if both tx_qos and rx_qos is set only in_path is set.
Is there something preventing this from being the case?
Should that be clarified in the documentation?

Reply:
> What does "default to HCI data path" mean?
Regarding the data path and HCI: The data path is either HCI, where data is then sent to the host via HCI, or a proprietary data path where the data isn't handled over HCI. The application can either provide a HCI path, a proprietary path or NULL to default to the HCI path. 

> The code sets up only one direction based off TX_QOS is NULL or not.
> So if both tx_qos and rx_qos is set only in_path is set.
The API only allows either `rx` or `tx` to be set for a broadcast ISO, so it shouldn't be confusing for the application. 

> Is there something preventing this from being the case?
The spec also only allows setting a single data path (for either IN or OUT depending on the BIS), so that's why we only set one. 

> Should that be clarified in the documentation?
No, I don't think that was very clear. Added more text for the `rx` and `tx` fields.